app/com/linkedin/drelephant/tony/util/TonyUtils.java

```diff
@@ -35,7 +35,7 @@ public static double getMaxMemoryBytesUsedForTaskType(Map<String, Map<Integer, T
     for (TonyTaskData taskData : taskMap.get(taskType).values()) {
       List<Metric> metrics = taskData.getMetrics();
       if (metrics == null) {
-        return -1;
+        break;
```

Comment:
should this be a continue? Is it possible for some tasks of the same type to have metrics and some not to have metrics?

Reply:
Ah, good catch. Yes, I think it should be a continue. Will add a test for this.

Reply:
Addressed in my latest commit.